### PR TITLE
Doc: Fix warnings / errors in cache architecture.

### PR DESCRIPTION
**Pull request by SolidWallOfCode**
Still more clean up needed, but at least with this the documentation generates cleanly.

### CODE REVIEW

**Comment by SolidWallOfCode** [comment]
[approve ci docs]

**Comment by SolidWallOfCode** [comment]
[approve ci doc]

**Comment by SolidWallOfCode** [comment]
[approve ci docs]

File: doc/developer-guide/architecture/cache-store.en.rst [hunk 1]
```diff
@@ -1,83 +0,0 @@
-.. Licensed to the Apache Software Foundation (ASF) under one
```
Comment (persiaAziz) [inline doc/developer-guide/architecture/cache-store.en.rst]: Intersting, I can not find the commit details where is says that this file is going to get deleted.

**Pull request merged**